### PR TITLE
refactor: improve error handling in LoadTxInfo

### PR DESCRIPTION
**Pull request by tiendn**
## Description

Improve error handling in LoadTxInfo

Replace panic with proper error returns in LoadTxInfo function to follow Go's
error handling conventions. This change:
- Adds input validation for empty transaction hash
- Returns descriptive error messages instead of panicking
- Wraps underlying errors with context
- Maintains backward compatibility with nil returns for not found cases

This is part of an ongoing effort to improve error handling patterns in the
codebase.
---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use
  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -561,20 +561,25 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
...
+// Returns nil if the transaction is not found.
+func (bs *BlockStore) LoadTxInfo(txHash []byte) (*cmtstore.TxInfo, error) {
+	if len(txHash) == 0 {
+		return nil, fmt.Errorf("cannot load tx info with empty hash")
+	}
+
 	bz, err := bs.db.Get(calcTxHashKey(txHash))
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to get tx info from db: %w", err)
```
Comment (rootulp) [inline store/store.go:572]: [question] should we add a unit test to check that this error is returned in certain cituations?

File: store/store.go [hunk 2]
```diff
@@ -561,20 +561,25 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
 }
 
 // LoadTxInfo loads the TxInfo from disk given its hash.
-func (bs *BlockStore) LoadTxInfo(txHash []byte) *cmtstore.TxInfo {
+// Returns nil if the transaction is not found.
+func (bs *BlockStore) LoadTxInfo(txHash []byte) (*cmtstore.TxInfo, error) {
+	if len(txHash) == 0 {
+		return nil, fmt.Errorf("cannot load tx info with empty hash")
+	}
```
Comment (rootulp) [inline store/store.go:568]: [question] should we add a unit test for this scenario?

File: store/store.go [hunk 3]
```diff
@@ -561,20 +561,25 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
...
+
 	bz, err := bs.db.Get(calcTxHashKey(txHash))
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to get tx info from db: %w", err)
 	}
 	if len(bz) == 0 {
-		return nil
+		return nil, nil
```
Comment (rootulp) [inline store/store.go:575]: [question] should we return an error in this situation? Also needs a unit test.

**Review by rootulp — COMMENTED** [review]
Overall I'm a fan of replacing `panic`s with errors so thanks! 

What motivated this PR? Is this related to a GitHub issue in the repo?

Before approving, it would be nice to add some unit tests that check for the new errors.

File: store/store.go (hunk 3 above)
Comment (tiendn) [reply store/store.go:575]: Yeah, I think it makes consistent.

**Comment by tiendn** [comment]
> What motivated this PR? Is this related to a GitHub issue in the repo?
Yeah,  i want to improve codebase

**Comment by tiendn** [comment]
Hi @rootulp can you review this PR

File: store/store.go [hunk 4]
```diff
@@ -561,20 +561,25 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
 }
 
 // LoadTxInfo loads the TxInfo from disk given its hash.
-func (bs *BlockStore) LoadTxInfo(txHash []byte) *cmtstore.TxInfo {
+// Returns nil if the transaction is not found.
+func (bs *BlockStore) LoadTxInfo(txHash []byte) (*cmtstore.TxInfo, error) {
+	if len(txHash) == 0 {
+		return nil, fmt.Errorf("cannot load tx info with empty hash")
```
Comment (rootulp) [inline store/store.go:567]:
```suggestion
		return nil, fmt.Errorf("cannot load tx info for empty txHash")
```

File: store/store.go (hunk 3 above)
Comment (rootulp) [reply store/store.go:575]: This still doesn't return an error. should it?

File: store/store_test.go [hunk 5]
```diff
@@ -776,3 +782,118 @@ func newBlock(hdr types.Header, lastCommit *types.Commit) *types.Block {
...
+	bs := NewBlockStore(dbm.NewMemDB())
+
+	testCases := []struct {
+		name          string
+		txHash        []byte
+		expectedError string
+		dbError       error 
+	}{
+		{
+			name:          "empty hash",
```
Comment (rootulp) [inline store/store_test.go:798]:
```suggestion
			name:          "empty txHash",
```

File: store/store_test.go [hunk 6]
```diff
@@ -776,3 +782,118 @@ func newBlock(hdr types.Header, lastCommit *types.Commit) *types.Block {
...
+			txHash:        []byte{},
+			expectedError: "cannot load tx info with empty hash",
+		},
+		{
+			name:          "non-existent tx",
+			txHash:        []byte("non_existent_hash"),
+			expectedError: "",  // no error, just nil result
```
Comment (rootulp) [inline store/store_test.go:805]: Should this return an error instead?

File: store/store_test.go [hunk 7]
```diff
@@ -776,3 +782,118 @@ func newBlock(hdr types.Header, lastCommit *types.Commit) *types.Block {
...
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.setup != nil {
+				tc.setup(bs.db)
+			}
+
+			// Both cases should return (nil, nil) to indicate "not found"
```
Comment (rootulp) [inline store/store_test.go:893]: Why use `(nil, nil)` to indicate "not found" instead of returning an error for not found?

**Review by rootulp — COMMENTED** [review]
Thanks for adding unit test! One last question about returning `(nil, nil)` instead of an error for the not found case.

**Comment by tiendn** [comment]
Hi @rootulp, thank you for reviewing my PR.
I've updated the error msg in case of not found. Please check

File: store/store_test.go [hunk 8]
```diff
@@ -776,3 +783,119 @@ func newBlock(hdr types.Header, lastCommit *types.Commit) *types.Block {
...
+	}{
+		{
+			name:          "Empty txHash",
+			txHash:        []byte{},
+			expectedError: "cannot load tx info for empty txHash",
+		},
+		{
+			name:          "Non-existent tx",
+			txHash:        []byte("non_existent_hash"),
+			expectedError: "transaction not found",  // no error, just nil result
```
Comment (rootulp) [inline store/store_test.go:806]:
```suggestion
			expectedError: "transaction not found",
```

File: store/store_test.go (hunk 7 above)
Comment (rootulp) [reply store/store_test.go:894]: This comment is no longer applicable
```suggestion
```

File: store/store_test.go [hunk 9]
```diff
@@ -776,3 +783,119 @@ func newBlock(hdr types.Header, lastCommit *types.Commit) *types.Block {
...
+}
+
+func (m *mockDBWithError) Get(key []byte) ([]byte, error) {
+	return nil, m.err
+}
+
+func TestLoadTxInfoNotFound(t *testing.T) {
+	bs := NewBlockStore(dbm.NewMemDB())
+
+	// Test cases for "not found" scenarios
```
Comment (rootulp) [inline store/store_test.go:868]: why is this unit test separate from the `TestLoadTxInfoErrors` function? Now that no tx found returns an error, it seems like we can consolidate the test functions.

**Review by rootulp — COMMENTED** [review]
Build is failing
